protos/flyteidl/admin/common.proto

```diff
@@ -25,8 +19,18 @@ message NamedEntityIdentifier {
     // The combination of project + domain + name uniquely identifies the resource.
     // +optional - in certain contexts - like 'List API', 'Launch plans'
     string name                  = 3;
+}
+
+message NamedEntityMetadata {
+    // Common description across all versions of the entity
+    // +optional
+    string description = 1;
+}
 
-    NamedEntityMetadata metadata = 4;
+// Composes a NamedEntityIdentifier with its associated metadata
+message NamedEntityIdentifierWithMetadata {
```

Comment:
thank you for refactoring! since this name is a bit long, what do you think about just "NamedEntity" ?

Reply:
I agree it's long. But since the original message is specifically called an "Identifier", I'm hesitant to remove that word in an effort to keep it consistent 😬

Reply:
sure, but the original message only handled identifier entities :). elsewhere we just refer to the underlying entity in message names rather than do something like ExecutionWithMetadata or WorkflowWithClosure

Reply:
I see what you're saying with the existing endpoint being named `NamedEntityIdentifierListRequest`. Perhaps, rather than deprecating the field within the endpoint we deprecate the endpoint altogether and just add a new one that better describes what this endpoint now returns like `NamedEntityListRequest`